help/en/html/doc/Technical/Patterns.shtml

```diff
@@ -92,9 +94,91 @@
+      the <code>Turnout</code> object registers itself as a change listener
+      when the <code>Sensor</code>'s state property changes, and updates
+      the <code>Turnout</code>'s <code>"KnownState"</code> property.</p>
+
+      <p>The available Bean properties are defined in the abstract base class
+      usually, for example <code>AbstractTurnout</code>
+      defines <code>"CommandedState"</code>,
+      <code>"KnownState"</code>, <code>"feedbackchange"</code>, <code>"locked"</code>
+      and some more at the time of this writing. These properties are not
+      system-dependent. Some of the properties are run-time only (e.g. state --
+      is the turnout thrown or closed?), while others (e.g. turnout feedback
+      mode) are configuration settings, selected by the user and saved between
+      sessions.</p>
+
+      <h3>Editing and saving NamedBeans</h3>
+
+      <p>NamedBeans are created and configured by the user using explicit
+      actions. Most of the UI for these actions is in
+      the <a href="http://jmri.org/JavaDoc/doc/jmri/jmrit/beantable/package-summary.html">jmri.jmrit.beantable</a>
+      package, using the generic <code>BeanTable{Frame,Pane,Model}</code>
+      classes specialized for the particular type, for example in the
+      <a href="http://jmri.org/JavaDoc/doc/jmri/jmrit/beantable/TurnoutTableAction.html"><code>TurnoutTableAction</code></a>
+      class. The configuration options present in the table and the edit dialog
+      are specific to the type (<code>Turnout</code>) but not the system.</p>
       
+      <p>The beans with the configured options are persisted into the
+      Configuration (and Panel) XML file when the user saves those. The
+      persistence is handled by the system- and object-specific ManagerXml
+      class, for
+      example <a href="http://jmri.org/JavaDoc/doc/jmri/jmrix/loconet/configurexml/LnTurnoutManagerXml.html">LnTurnoutManagerXml</a>
+      or <a href="http://jmri.org/JavaDoc/doc/jmri/jmrix/openlcb/configurexml/OlcbTurnoutManagerXml.html">OlcbTurnoutManagerXml</a>,
+      which heavily rely on shared code
+      in <a href="http://jmri.org/JavaDoc/doc/jmri/managers/configurexml/AbstractTurnoutManagerConfigXML.html">AbstractTurnoutManagerConfigXML</a>,
+      but can introduce system-specific functionality and work together with
+      the system- and object-specific manager
+      (e.g. <a href="http://jmri.org/JavaDoc/doc/jmri/jmrix/openlcb/OlcbTurnoutManager.html">OlcbTurnoutManager</a>)
+      to achieve this.</p>
+
+      <p>The base class handles persisting the user settings that were entered
+      via the BeanTable.</p>
+
+      <h3>System-specific properties</h3>
+      
+      <p>Adding a system-specific property requires using a generic API,
+      because the code in the <code>jmrit.beantable</code>
+      package <a href="IntroStructure.shtml">cannot depend</a> on the
+      jmrix.system-specifc packages. All NamedBeans have
```

Comment:
Thanks for add this, it's very clear & useful.  

Just noticed a typo on this line: "specifc". Not a big deal.

Reply:
fixed